tdms/src/utils.cpp

```diff
@@ -0,0 +1,23 @@
+#include "cstdio"
+#include "string"
+#include "cstring"
+#include "stdexcept"
+
+
+using namespace std;
+
+
+void assert_can_open_file(const char* filename, const char* mode){
+
+  auto file = fopen(filename, mode);
+
+  if(file == nullptr){
+    throw runtime_error("Unable to open file " + string(filename));
+  }
+
+  fclose(file);
+}
+
+bool are_equal(const char* a, const char* b){
```

Comment:
```suggestion
bool are_equal_str(const char* a, const char* b){
```
Is a clearer function name?

Reply:
no need for 🦆s here, imo